### PR TITLE
Check disk space

### PR DESCRIPTION
**Pull request by XX-Yin**
### Pull Request instructions:
- Please follow the [update protocol](https://github.com/AllenNeuralDynamics/aind-behavior-blog/wiki/Software-Update-Procedures)
- Answer the questions below in detail. Your responses will be emailed to experimenters. 
- If the experimenters must do anything new, provide detailed step by step instructions on the [wiki](https://github.com/AllenNeuralDynamics/aind-behavior-blog/wiki)
- If computer maintainers need to manually update anything, provide detailed step by step instructions
- Use markdown syntax in order for your comments to be rendered reliably in the email: "1." instead of "1)", use four spaces for indents.
- If you use the keyword "skip email" in the title, it will skip the email updates
- Merges from "develop" into "production_testing" should use the keyword "production merge" in the title for reliable indexing of updates
- Merges from "production_testing" into "main" should use the keyword "update main"
  
### Describe changes:
Displays disk space and warns users of low disk space when the available space is less than 100G or less than 1% of the total space.

image

### What issues or discussions does this update address?
Avoid disk fullness
Related: https://github.com/orgs/AllenNeuralDynamics/projects/47/views/7?pane=issue&itemId=71011651

### Describe the expected change in behavior from the perspective of the experimenter
No

### Describe any manual update steps for task computers
No

### Was this update tested in 446/447?
Tested on my own computer.

### CODE REVIEW

File: src/foraging_gui/Foraging.py [hunk 1]
```diff
@@ -176,6 +179,16 @@ def _load_rig_metadata(self):
         self.latest_rig_metadata_file = rig_json_file 
         self.Metadata_dialog._SelectRigMetadata(self.latest_rig_metadata_file)
 
+    def _show_disk_space(self):
+        '''Show the disk space of the current computer'''
+        total, used, free = shutil.disk_usage(self.default_saveFolder)
+        self.diskspace.setText(f"Used space: {used/1024**3:.2f}GB    Free space: {free/1024**3:.2f}GB")
+        self.DiskSpaceProgreeBar.setValue(int(used/total*100))
+        if free/1024**3 < 100 or used/total > 0.9:
+            self.DiskSpaceProgreeBar.setStyleSheet("QProgressBar::chunk {background-"+self.default_warning_color+"}")
```
Comment (alexpiet) [inline src/foraging_gui/Foraging.py:188]: Can we additionally log an error if the disk space is low?

**Comment by alexpiet** [comment]
- Does the bar update, or just show the status when the GUI opens?
- The bar is cool, but if it doesn't update, maybe we should just have a pop up window to alert the user of low disk space?

**Comment by XX-Yin** [comment]
> * Does the bar update, or just show the status when the GUI opens?
> * The bar is cool, but if it doesn't update, maybe we should just have a pop up window to alert the user of low disk space?

The bar will update every trial

File: src/foraging_gui/Foraging.py (hunk 1 above)
Comment (XX-Yin) [reply src/foraging_gui/Foraging.py:188]: Good point. Added.

**Review by alexpiet — APPROVED** [review]
Looks great!

**Pull request merged**